File: app/src/main/java/com/github/marlonlom/demos/barcodescans/ViewPagerAdapter.java
package com.github.marlonlom.demos.barcodescans;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.ArrayList;

/**
 * Viewpager adapter class
 *
 * @author marlonlom
 */
final class ViewPagerAdapter extends FragmentPagerAdapter {

    private ArrayList<Fragment> mFragments;

    ViewPagerAdapter(FragmentManager fm) {
        super(fm);
        mFragments = new ArrayList<>(2);
    }

    @Override
    public CharSequence getPageTitle(int position) {
        if (position >= 0 && position < 2) {
            return "Dummy,Camera".split(",")[position];
        }
        return null;
    }

    @Override
    public Fragment getItem(int position) {
        return mFragments.get(position);
    }

    @Override
    public int getCount() {
        return mFragments.size();
    }

    void addFragment(Fragment fragment) {
        mFragments.add(fragment);
    }
}

File: app/src/main/java/com/github/marlonlom/demos/barcodescans/ViewPagerAdapter.kt
package com.github.marlonlom.demos.barcodescans

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter

import java.util.ArrayList

/**
 * Viewpager adapter class

 * @author marlonlom
 */
internal class ViewPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

    private val mFragments: ArrayList<Fragment>

    init {
        mFragments = ArrayList<Fragment>(2)
    }

    override fun getPageTitle(position: Int): CharSequence? {
        if (position >= 0 && position < 2) {
            return "Dummy,Camera".split(",".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()[position]
        }
        return null
    }

    override fun getItem(position: Int): Fragment {
        return mFragments[position]
    }

    override fun getCount(): Int {
        return mFragments.size
    }

    fun addFragment(fragment: Fragment) {
        mFragments.add(fragment)
    }
}
